### PR TITLE
feat: transaction fail on timeout

### PR DESCRIPTION
**Pull request by skeremidchiev**
#Linear Ticket

[LINEAR TASK LIQ-374](https://linear.app/liquality/issue/LIQ-374/when-transactions-are-done-successive-then-some-transactions-are)

### CODE REVIEW

File: src/store/actions/performNextAction/send.js [hunk 1]
```diff
@@ -12,6 +14,17 @@ async function waitForConfirmations ({ getters, dispatch }, { transaction, netwo
   )
   try {
     const tx = await client.chain.getTransactionByHash(transaction.txHash)
+
+    // In case transaction doesn't exist: wait for 30 minutes then return failed state
+    // Could happen on Dropped&Replaced
+    if (!tx && hasTimedOut(transaction)) {
+      return {
+        endTime: Date.now(),
+        status: TxStatus.Failed
```
Comment (tima-t) [inline src/store/actions/performNextAction/send.js:23]: TxStatus.Failed. - what is the value of this. 
Is it one of the supported states - FAILED/ REFUND etc

File: src/swaps/liquality/LiqualitySwapProvider.js [hunk 2]
```diff
@@ -272,7 +273,11 @@ class LiqualitySwapProvider extends SwapProvider {
 
     try {
       const tx = await fromClient.chain.getTransactionByHash(swap.fromFundHash)
-
+      if (!tx && hasTimedOut(swap)) {
+        return {
+          status: 'REFUNDED'
```
Comment (tima-t) [inline src/swaps/liquality/LiqualitySwapProvider.js:278]: I guess that you should have 

endTime property here

similar to
          endTime: Date.now(),
          status: 'FAILED'
     
Might be important for the timeline diagram

File: src/swaps/liquality/LiqualitySwapProvider.js [hunk 3]
```diff
@@ -347,7 +352,11 @@ class LiqualitySwapProvider extends SwapProvider {
     const toClient = this.getClient(network, walletId, swap.to, swap.toAccountId)
 
     const tx = await toClient.chain.getTransactionByHash(swap.toFundHash)
-
+    if (!tx && hasTimedOut(swap)) {
+      return {
+        status: 'REFUNDED'
```
Comment (tima-t) [inline src/swaps/liquality/LiqualitySwapProvider.js:357]: same here for the  endTime property

File: src/swaps/liquality/LiqualitySwapProvider.js [hunk 4]
```diff
@@ -392,7 +401,11 @@ class LiqualitySwapProvider extends SwapProvider {
 
     try {
       const tx = await toClient.chain.getTransactionByHash(swap.toClaimHash)
-
+      if (!tx && hasTimedOut(swap)) {
+        return {
+          status: 'REFUNDED'
```
Comment (tima-t) [inline src/swaps/liquality/LiqualitySwapProvider.js:406]: same here for the  endTime property

File: src/swaps/liquality/LiqualitySwapProvider.js [hunk 5]
```diff
@@ -421,7 +434,11 @@ class LiqualitySwapProvider extends SwapProvider {
     const fromClient = this.getClient(network, walletId, swap.from, swap.fromAccountId)
     try {
       const tx = await fromClient.chain.getTransactionByHash(swap.refundHash)
-
+      if (!tx && hasTimedOut(swap)) {
+        return {
+          status: 'REFUNDED'
```
Comment (tima-t) [inline src/swaps/liquality/LiqualitySwapProvider.js:439]: same here for the  endTime property

File: src/store/actions/performNextAction/send.js (hunk 1 above)
Comment (skeremidchiev) [reply src/store/actions/performNextAction/send.js:23]: Values is 'FAILED'

**Pull request closed without merging**